src/main/kotlin/com/esops/repository/OrderRepository.kt

```diff
@@ -0,0 +1,67 @@
+package com.esops.repository
+
+import com.esops.entity.BuyOrderComparator
+import com.esops.entity.Order
+import com.esops.entity.OrderType
+import com.esops.entity.SellOrderComparator
+import jakarta.inject.Singleton
+import java.math.BigInteger
+import java.util.*
+
+@Singleton
+class OrderRepository {
+
+    private var orders = HashMap<String, HashMap<Long, Order>>()
+    private var buyOrderQueue = PriorityQueue(BuyOrderComparator)
+    private var sellOrderQueue = PriorityQueue(SellOrderComparator)
+    private var orderIDCounter: Long = 0
+
+    fun incrementOrderIDCounter() {
+        orderIDCounter++
+    }
+
+    fun getOrderIDCounter(): Long {
+        return orderIDCounter
+    }
+
+    fun initializeOrderMapIfEmpty(username: String) {
+        if (orders[username].isNullOrEmpty())
+            orders[username] = HashMap()
+    }
+
+    fun getOrderByUsername(username: String): List<Order> {
+        return orders[username]!!.values.toList()
+    }
+
+    fun addOrder(username: String, order: Order) {
+        orders[username]?.set(orderIDCounter, order)
+    }
+
+    fun getOrderQueue(type: OrderType): PriorityQueue<Order> {
+        if(type == OrderType.BUY)   return buyOrderQueue
+        return sellOrderQueue
+    }
+
+    fun setOrderQueue(type: OrderType, orderQueue: PriorityQueue<Order>) {
+        if(type == OrderType.BUY)   buyOrderQueue = orderQueue
+        if(type == OrderType.SELL)   sellOrderQueue = orderQueue
+    }
+
+    fun addToOrderQueue(type: OrderType, order: Order) {
+        if(type == OrderType.BUY)   buyOrderQueue.add(order)
+        if(type == OrderType.SELL)   sellOrderQueue.add(order)
+    }
+
+    fun cleanQueue() {
+        buyOrderQueue.removeIf { it.remainingQuantity == BigInteger("0") }
+        sellOrderQueue.removeIf { it.remainingQuantity == BigInteger("0") }
+    }
+
+    fun clearOrders() {
+        orders = HashMap()
+        orderIDCounter = 0
+        buyOrderQueue.clear()
```

Comment:
clear? What is clean orders - clearly you are not just `cleaningOrders`

Reply:
Only using in controller tests, unit tests changed